### PR TITLE
support changefeed stop/pause/remove operation

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

We have no management to a running changefeed, such as pause replication, resume replication or remove this changefeed from a CDC cluster.

### What is changed and how it works?

- Treat changefeed as resource both in owner and processor, so if a DDL executed to downstream with failure, we don't exit the whole owner, but only pause the changefeed replication.
- Introduce the `AdminJob` concept, which can be processed by the owner only. We can use this feature by HTTP API now.
- In both `Stop` and `Remove` job, we release all the TiKV gRPC connections and stop the replication. The difference between these two jobs is the `Stop` job will preserve the `Changefeed` information, so we can resume the replication of this changefeed

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -322,6 +323,9 @@ type ownerImpl struct {
 	captureWatchC      <-chan *CaptureInfoWatchResp
 	cancelWatchCapture func()
 	captures           map[model.CaptureID]*model.CaptureInfo
+
+	jobs     []*model.AdminJob
```
Comment (leoppro) [inline cdc/owner.go]: call it `adminJobs` to avoid confusion with `ddlJob`?

File: cdc/http_handler.go [hunk 2]
```diff
@@ -15,29 +15,64 @@ package cdc
...
 	"net/http"
+	"strconv"
 
 	"github.com/coreos/etcd/clientv3/concurrency"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+const (
+	opVarAdminJob     = "job"
```
Comment (leoppro) [inline cdc/http_handler.go]:
```suggestion
	opVarAdminJob     = "admin-job"
```

File: cdc/http_handler.go [hunk 3]
```diff
@@ -15,29 +15,64 @@ package cdc
...
 	if req.Method != http.MethodPost {
 		writeError(w, http.StatusBadRequest, errors.New("this api only supports POST method"))
 		return
 	}
 	err := s.capture.ownerManager.ResignOwner(req.Context())
+	handleOwnerResp(w, err)
+}
+
+func (s *Server) handleChangefeedAdmin(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
```
Comment (suzaku) [inline cdc/http_handler.go]: We can restrict HTTP methods by adding a call to `Methods("POST")` after `HandleFunc`.

File: cdc/owner.go [hunk 4]
```diff
@@ -322,6 +323,9 @@ type ownerImpl struct {
 	captureWatchC      <-chan *CaptureInfoWatchResp
 	cancelWatchCapture func()
 	captures           map[model.CaptureID]*model.CaptureInfo
+
+	adminJobs     []*model.AdminJob
```
Comment (suzaku) [inline cdc/owner.go]: No need to use pointer for small structs.

File: cdc/http_handler.go (hunk 3 above)
Comment (amyangfei) [reply cdc/http_handler.go]: default http `serverMux` does not provide `methodMatcher`

Comment (suzaku) [reply cdc/http_handler.go]: We can use `gorilla/mux` later.

File: cdc/model/changefeed.go [hunk 5]
```diff
@@ -33,8 +32,12 @@ type ChangeFeedDetail struct {
 	// Start sync at this commit ts if `StartTs` is specify or using the CreateTime of changefeed.
 	StartTs uint64 `json:"start-ts"`
 	// The ChangeFeed will exits until sync to timestamp TargetTs
-	TargetTs uint64          `json:"target-ts"`
-	Info     *ChangeFeedInfo `json:"-"`
+	TargetTs uint64 `json:"target-ts"`
+	// ChangeFeed can be resumed from stop state, this field is used to trigger a watch event in each capture
+	ResumeTs uint64 `json:"resume-ts"`
```
Comment (suzaku) [inline cdc/model/changefeed.go]: Is it possible to use `checkpointTs` to avoid introducing a similar concept?

Comment (amyangfei) [reply cdc/model/changefeed.go]: removed, PTAL

**Pull request merged**